ndonnx/_typed_array/funcs.py

```diff
@@ -64,9 +64,10 @@ def _infer_dtype(
     elif isinstance(val, Sequence):
         return _infer_sequence(val)
     elif isinstance(val, np.generic):
+        if val.dtype.kind in "Mm":
+            return datetime.from_numpy_dtype(val.dtype)
         return onnx.from_numpy_dtype(val.dtype)
```

Comment:
Can we consolidate this into one function for determining the ndonnx data type from the numpy dtype?